chat-android/src/main/java/com/ably/chat/Utils.kt

```diff
@@ -217,3 +236,64 @@ internal class LatestJobExecutor {
         }
     }
 }
+
+/**
+ * A custom implementation of a `MutableSharedFlow` that supports optional awaitable emissions.
+ *
+ * This class allows emitting values with an optional `CompletableDeferred` to track the completion
+ * of processing for each emitted value. It is useful in scenarios where you need to ensure that
+ * all consumers have processed the emitted values before proceeding.
+ *
+ * @param T The type of values emitted by the shared flow.
+ * @property logger A logger instance used to log errors during collection.
+ */
+@Suppress("OutdatedDocumentation")
```

Comment:
Why `@Suppress("OutdatedDocumentation")`?

Reply:
Ohh, sorry, this was temporary. I tried suppressing it to check all tests work on CI.
Fixed as a part of -> https://github.com/ably/ably-chat-kotlin/pull/124/commits/078950b7942309cac6ec10c7436abef0d3234c39